app/editor/src/features/content/form/ClipDirectoryFilter.tsx

```diff
@@ -0,0 +1,33 @@
+import { IconButton } from 'components/form';
+import React from 'react';
+import { Row, Text } from 'tno-core';
+
+interface IClipDirectoryFilterProps {
+  setGlobalFilter: (filterValue: any) => void;
+}
+
+export const ClipDirectoryFilter: React.FC<IClipDirectoryFilterProps> = ({ setGlobalFilter }) => {
+  const [filter, setFilter] = React.useState<string>('');
```

Comment:
For the issue of the filter containing the appropriate text but not applying after create clip is clicked, you could try adding a `useEffect` here that makes sure the `globalFilter` and the filter align. You could also then remove the `setGlobalFilter()` calls and just have them handled in the `useEffect`.  You would just have to add `globalFilter` in this context.

eg) 
In your interface you could add:
` state: { globalFilter: any };` and make sure to add it on line 9 as well

Then try:

`useEffect(() => {
  if(filter != globalFilter) {
    setGlobalFilter(filter)
  }
}, [filter, globalFilter, setGlobalFilter])`

It seems like the two get out of sync at some point during create clip.